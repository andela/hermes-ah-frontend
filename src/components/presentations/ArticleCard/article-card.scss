@import '../../../assets/stylesheets/index.scss';
.article-card-item{
    .author-name {
        color: black;
        font-weight: 600
    }
    
    .category {
        text-transform: uppercase;
        margin: 0;
        color: #0E71EB;
    } 

    .art-header {
        color: #0E71EB;
    }

    .art-header:hover {
        opacity: 0.8;
    }

    .description {
        color: #000;
        cursor: pointer;
    }

    .description:hover {
        color: #004190 !important;
    }

    a.header {
        text-transform: capitalize;
    }

}
.ui.items>.item>.image>img {
    height: 169px;
}
.twelve.wide.computer.sixteen.wide.mobile.column{
    height: 69vh;
    overflow-y:auto;
}

.twelve.wide.computer.sixteen.wide.mobile.column::-webkit-scrollbar-track
{
	background-color: #ffffff;
}

.twelve.wide.computer.sixteen.wide.mobile.column::-webkit-scrollbar-thumb
{
	background-color: #ffffff;
}
.twelve.wide.computer.sixteen.wide.mobile.column:hover::-webkit-scrollbar-thumb
{
	border-radius: 10px;
	-webkit-box-shadow: inset 0 0 6px rgba(255, 255, 255, 0.3);
	background-color: #8f8f8f;
}
