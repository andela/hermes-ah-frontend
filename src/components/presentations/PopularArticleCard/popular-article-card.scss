
.article-section-hmp {
    .popular-articles-card {
        margin-top: 10px;
    }
    
    .author-name {
        color: black;
        font-weight: 600
    }
    
    .display-article-card {
        margin-top: 10px;
     
    }
    .header-num {
            color: #959595;
            margin: 0 10px;
            line-height: 0.6em;
    }
    .ui.grid {
        margin: 0;
    }
    .ui.small.image{
        width: 250px;
        margin: 0 10px;
    }

    i.far.fa-thumbs-up {
        font-size: 18px;
        cursor: pointer;
    }

    i.far.fa-thumbs-up:hover {
        color: #0E71EB;
        transform: scale(1.1)
    }

    .ui.button {
        background: #ffffff;
    }

    .ui.button:hover {
        background: #ffffff;
    }

    .ui.button:active {
        background: #ffffff;
    }

}
