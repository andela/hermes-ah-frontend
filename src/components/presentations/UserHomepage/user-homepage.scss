.article-section-hmp.cont {
  .twelve.wide.computer.sixteen.wide.mobile.column {
    height: 55vh;
  }
  hr.line-brk {
    margin: 40px 0;
    background-color: #86868688;
    height: 0;
  }
  .ui.items.section-2 {
    width: 100%;
    margin: 0 auto;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    .b-cont-wrap {
      width: 50%;
    }
  
    .a-cont-wrap {
      width: 45%;
    }
    .c-cont-wrap {
      width: 25%;
      display: flex;
      align-items: flex-end;
      flex-direction: column;
    }
  
    .c-sectn {
      display: flex;
      flex-direction: column;
      width: 90%;
      align-items: center;
    }
    .a-sectn {
      display: flex;
      flex-direction: column;
      width: 100%;
      margin-bottom: 40px;
      display: flex;
      flex-direction: column;
      margin-bottom: 40px;
      justify-content: center;
      align-items: center;
      .a-image {
        width: 100%;
        img {
          width: 100%;
          height: 223px !important;
        }
      }
    }
    .author {
      color: #000;
      font-weight: 600;
      margin: 2px;
    }
    .date-start {
      display: flex;
      .stats {
        margin-left: 15px;
      }
    }
    .description, .date, .stats {
      color: #555454;
    }
    .c-image {
      width: 100%;
      img {
        height: 210px !important;
        width: 100%;
      }
    }
    .b-image {
      float: right;
      width: 40%;
      img {
        width: 100%;
        height: 103px !important;
      }
    }
    .b-sectn {
      display: flex;
      flex-direction: row;
      width: 100%;
      margin-bottom: 10px;
    }
  }
  .art-title {
    font-size: 19px;
    font-weight: bold;
    margin: 10px 0;
    text-transform: capitalize;
  }
  .content.b-content {
    width: 100%;
  }
  .b-art-title {
    font-size: 19px;
    font-weight: bold;
    margin: 10px 0;
    text-transform: capitalize;
  }
  
  p.desc-txt {
    margin-bottom: 10px;
  }
  
  .ui.container.wrap-A-B {
    width: 75%;
    display: flex;
    justify-content: space-between;
  }
}
