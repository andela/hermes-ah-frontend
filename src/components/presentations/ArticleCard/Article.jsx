import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Item, Header } from 'semantic-ui-react';
import './article-card.scss';

const ArticleCard = ({
  category,
  title,
  author,
  date,
  read,
  paragraph,
  image,
  articleId,
}) => (
  <Item.Group className="article-card-item">
    <Header as="h3" className="category">
      {category}
    </Header>
    <Item>
      <Item.Content>
        <br />
        <Link to={`/article/${articleId}`} className="link">
          <h3 className="art-header">{title}</h3>
          <Item.Description>
            {`${paragraph.charAt(0).toUpperCase() +
              paragraph.slice(1).substring(0, 200)}...`}
          </Item.Description>
        </Link>
        <Item.Extra>
          <p className="author-name">{author}</p>
        </Item.Extra>
        <Item.Extra>
          <p>
            {new Date(date).toDateString()}
            &nbsp; &nbsp; &nbsp;
            {read}
          </p>
        </Item.Extra>
      </Item.Content>
      <Item.Image
        size="small"
        src={
          image ||
          'https://res.cloudinary.com/dnch08bzc/image/upload/v1557150721/science-1182713_1280.jpg'
        }
      />
    </Item>
  </Item.Group>
);

ArticleCard.defaultProps = {
  image:
    'https://res.cloudinary.com/dnch08bzc/image/upload/v1557150721/science-1182713_1280.jpg',
};

ArticleCard.defaultProps = {
  date: '10/05/2019, 08:49:38',
  articleId: '',
};

ArticleCard.propTypes = {
  articleId: PropTypes.string,
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  date: PropTypes.string,
  read: PropTypes.string.isRequired,
  paragraph: PropTypes.string.isRequired,
  image: PropTypes.string,
  category: PropTypes.string.isRequired,
};

export default ArticleCard;
