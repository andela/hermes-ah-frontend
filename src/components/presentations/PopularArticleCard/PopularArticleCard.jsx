import React from 'react';
import PropTypes from 'prop-types';
import { Item, Button } from 'semantic-ui-react';
import { decodeToken } from '../../../utils/authService';
import './popular-article-card.scss';

const PopularArticleCard = ({ title, author, date, likes, num, onClick }) => (
  <Item.Group>
    <Item>
      <Item.Header className="header-num" as="h1">
        {num}
      </Item.Header>
      <Item.Content>
        <Item.Header as="a">{title}</Item.Header>
        <Item.Extra>
          <p className="author-name">{author}</p>
        </Item.Extra>
        <Item.Extra>
          <div>
            {new Date(date).toDateString()}
            &nbsp; &nbsp; &nbsp;
            {decodeToken() && (
              <React.Fragment>
                <Button onClick={onClick}>
                  <i className="far fa-thumbs-up" />
                </Button>
                <p>
                  &nbsp;
                  {likes}
                </p>
              </React.Fragment>
            )}
          </div>
        </Item.Extra>
      </Item.Content>
    </Item>
  </Item.Group>
);

PopularArticleCard.defaultProps = {
  date: '10/05/2019, 08:49:38',
  likes: 0,
};

PopularArticleCard.propTypes = {
  num: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  date: PropTypes.string,
  likes: PropTypes.number,
  onClick: PropTypes.func.isRequired,
};

export default PopularArticleCard;
