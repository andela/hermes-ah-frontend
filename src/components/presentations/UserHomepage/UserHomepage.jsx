import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Item, Container, Grid } from 'semantic-ui-react';
import ArticleCard from '../ArticleCard/Article';
import PopularArticleCard from '../PopularArticleCard/PopularArticleCard';
import './user-homepage.scss';

class UserHomepage extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount = () => {
    const { getAllArticles: fetchArticles } = this.props;
    fetchArticles();
  };

  render() {
    const { articles } = this.props;
    const { articleData } = articles;

    const popularArticles = articleData.sort(
      (a, b) => b.likes_count - a.likes_count
    );

    const sectionArandomArticle = articleData.sort(() => Math.random() - 0.5);

    const popularArticlesLimit = popularArticles.slice(0, 3);

    const sectionBrandomArticle = articleData.sort(() => Math.random() - 0.5);

    const sectionBdivA = sectionBrandomArticle.slice(0, 1);
    const sectionBdivB = sectionBrandomArticle.slice(1, 4);
    const sectionBdivC = sectionBrandomArticle.slice(4, 5);

    return (
      <div className="article-section-hmp cont">
        <Grid>
          <Grid.Row>
            <Grid.Column computer={12} mobile={16}>
              <Container className="display-article-card">
                {sectionArandomArticle.map(item => (
                  <ArticleCard
                    key={item.id}
                    category={item.category}
                    title={item.title}
                    author={`${item.author.first_name} ${
                      item.author.last_name
                    }`}
                    date={item.createdAt}
                    read={`${item.reading_time} min read`}
                    paragraph={item.abstract}
                    image={item.image_url}
                  />
                ))}
              </Container>
            </Grid.Column>
            <Grid.Column computer={4} mobile={16}>
              <h2 className="custom-center">POPULAR POST</h2>
              <Container className="popular-articles-card">
                {popularArticlesLimit.map(elem => (
                  <PopularArticleCard
                    key={elem.id}
                    num={`0${popularArticlesLimit.indexOf(elem) + 1}`}
                    title={elem.title}
                    author={`${elem.author.first_name} ${
                      elem.author.last_name
                    }`}
                    date={elem.createdAt}
                    likes={elem.likes_count}
                  />
                ))}
              </Container>
            </Grid.Column>
          </Grid.Row>
        </Grid>
        <hr className="line-brk" />
        <Item.Group className="section-2">
          <Container className="wrap-A-B">
            <div className="a-cont-wrap">
              {sectionBdivA.map(article => (
                <Item className="a-sectn" key={article.id}>
                  {article.image_url ? (
                    <Item.Image
                      className="a-image"
                      size="small"
                      src={article.image_url}
                    />
                  ) : null}

                  <Item.Content className="a-content">
                    <Item.Header className="art-title">
                      {article.title}
                    </Item.Header>
                    <Item.Description>
                      <p className="description">
                        {`${article.abstract.substring(0, 200)}...`}
                      </p>
                      <p className="author">
                        {`${article.author.first_name} ${
                          article.author.last_name
                        }`}
                      </p>
                      <p className="date">
                        {new Date(article.createdAt).toDateString()}
                      </p>
                    </Item.Description>
                  </Item.Content>
                </Item>
              ))}
            </div>

            <div className="b-cont-wrap">
              {sectionBdivB.map(article => (
                <Item className="b-sectn" key={article.id}>
                  <Item.Content className="b-content">
                    <Item.Header className="b-art-title">
                      {article.title}
                    </Item.Header>
                    <Item.Description>
                      <div className="description">
                        {article.image_url ? (
                          <Item.Image
                            className="b-image"
                            size="small"
                            src={article.image_url}
                          />
                        ) : null}
                        <p className="desc-txt">
                          {`${article.abstract.substring(0, 100)}...`}
                        </p>
                      </div>
                      <p className="author">
                        {`${article.author.first_name} ${
                          article.author.last_name
                        }`}
                      </p>
                      <div className="date-start">
                        <p className="date">
                          {new Date(article.createdAt).toDateString()}
                        </p>
                        <p className="stats">
                          {`${article.reading_time} min read`}
                        </p>
                      </div>
                    </Item.Description>
                  </Item.Content>
                </Item>
              ))}
            </div>
          </Container>
          <div className="c-cont-wrap">
            {sectionBdivC.map(article => (
              <Item className="c-sectn" key={article.id}>
                {article.image_url ? (
                  <Item.Image
                    className="c-image"
                    size="small"
                    src={article.image_url}
                  />
                ) : null}

                <Item.Content className="a-content">
                  <Item.Header className="art-title">
                    {article.title}
                  </Item.Header>
                  <Item.Description>
                    <p className="description">
                      {`${article.abstract.substring(0, 200)}...`}
                    </p>
                    <p className="author">
                      {`${article.author.first_name} ${
                        article.author.last_name
                      }`}
                    </p>
                    <div className="date-start">
                      <p className="date">March 13 2019</p>
                      <p className="stats">7 min read</p>
                    </div>
                  </Item.Description>
                </Item.Content>
              </Item>
            ))}
          </div>
        </Item.Group>
      </div>
    );
  }
}

UserHomepage.propTypes = {
  articles: PropTypes.shape({}).isRequired,
  getAllArticles: PropTypes.func.isRequired,
};

export default UserHomepage;
