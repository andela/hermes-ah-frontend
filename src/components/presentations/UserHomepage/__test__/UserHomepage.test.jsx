import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import UserHomepage from '../UserHomepage';
import PopularArticleCard from '../../PopularArticleCard/PopularArticleCard';
import ArticleCard from '../../ArticleCard/Article';
import mock from '../../../../utils/testMocks';

const { articleCardProps, articleReducer } = mock;

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const store = mockStore({});

const popularArticleCardProps = {
  title: 'title',
  author: 'author',
  date: '2017-11-25T12:34:56z',
  likes: 2,
  num: '2',
};

const UserHomePageComponent = (
  <Provider store={store}>
    <UserHomepage {...articleReducer} />
  </Provider>
);

describe('userHomePage component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(UserHomePageComponent);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    const wrapper = shallow(<UserHomepage {...articleReducer} />);
    expect(wrapper.find('Item'));
    expect(wrapper.instance().onLikeClick(1)).toMatchSnapshot();
  });

  it('should render ArticleCard without crashing', () => {
    const wrapper = shallow(<ArticleCard {...articleCardProps} />);
    expect(wrapper.find('div'));
  });

  it('should render PopularArticleCard without crashing', () => {
    const wrapper = shallow(
      <PopularArticleCard {...popularArticleCardProps} />
    );
    expect(wrapper.find('div'));
  });
});
