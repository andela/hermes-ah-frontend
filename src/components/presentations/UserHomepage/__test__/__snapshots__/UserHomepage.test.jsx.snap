// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userHomePage component should match snapshot 1`] = `ShallowWrapper {}`;

exports[`userHomePage component should render without crashing 1`] = `undefined`;
